### ISSUE TITLE
SC2115 warns even when there's nothing to expand

### ISSUE THREAD

**Issue by mrc0mmand** [koalaman/shellcheck#2833]
#### For bugs
- Rule Id: SC2115
- My shellcheck version (`0.9.0` on Fedora Rawhide): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
Hey!

Today, whilst debugging, I was caught off guard by this shellcheck warning:

```sh
#!/bin/bash

rm -rf /boot/*
```

```
# shellcheck foo.sh 

In foo.sh line 3:
rm -rf /boot/*
       ^-----^ SC2115 (warning): Use "${var:?}" to ensure this never expands to /boot/* .
```

since there's no variable to expand, hence nothing to fix (unless I'm missing something).

**Comment by aeiplatform** [comment]
This would ve valid only for absolute paths and not even those.

Shellcheck is not shell interpreter!
Example:
```
touch idc
ls *
```
How could it know that there is something?
When files are created at run time.
